### ISSUE TITLE
Typescript: getMany() parameter mismatch with IVectorView and IVector 

### ISSUE THREAD

**Issue by jechternach** [microsoft/react-native-winrt#255]
[getMany() expects 2 function arguments in winrt](https://learn.microsoft.com/en-us/uwp/api/windows.foundation.collections.ivectorview-1.getmany?view=winrt-22621) with a function signature: `uint GetMany(uint startIndex, T[] items)`

It looks like in C++ codegen `getMany(int, T[]) `also expects two arguments, while the corresponding TypeScript codegen expects getMany to only take one argument. 

There exist the logic / ability to create structs to return back to TS but it is missing for getMany, because there existing special casing in the codegen for `indexOf` which natively has a function signature similar to getMany of `bool IndexOf(value, [out] uint index)`

**Comment by dunhor** [comment]
In this case, the TypeScript generation is incorrect. Arrays in WinRT are special in the sense that there's two different ways they can be handled. There's "output arrays" where the callee allocates and fills in the array and "fill arrays" where the caller allocates the array and the callee fills it in. The difference in metadata - I believe - is one is `[out] T[]` and the other is `[out] ref T[]`. I don't recall which one is which (I think `ref` means "fill"). If it's a fill array, the expected pattern is that the argument appears in the argument list (buffer provided by caller). The `GetMany` function accepts a fill array parameter.

TL;DR: The C++ codegen is correct. The TypeScript codegen likely doesn't fully understand the nuance of fill arrays

**Comment by dunhor** [comment]
Duplicate of #175
